apps/alerts/lib/cache/bus_stop_change_s3.ex

```diff
@@ -0,0 +1,119 @@
+defmodule Alerts.Cache.BusStopChangeS3 do
+  @moduledoc """
+  Functionality for writing to/reading from S3. Takes %Alert{} converts them into %HistoricalAlert{} before saving.
+  """
+  use RepoCache, ttl: :timer.hours(24)
+  require Logger
+  alias Alerts.{Alert, HistoricalAlert}
+
+  @ex_aws Application.compile_env(:alerts, [:aws_client], ExAws)
+  @ex_aws_s3 Application.compile_env(:alerts, [:s3_client], ExAws.S3)
+  @bucket "mbta-dotcom"
+  @bucket_prefix "bus-stop-change"
+
+  @doc """
+  Arguments for the Alerts.Cache.Fetcher process, Fetches current bus alerts
+  hourly, sending the parsed alerts to copy_alerts_to_s3/2
+  """
+  def fetcher_opts do
+    [
+      api_mfa:
+        {V3Api.Alerts, :all,
+         [
+           [
+             "filter[route_type]": 3,
+             "filter[datetime]": "NOW"
+           ]
+         ]},
+      repeat_ms: 3_600_000,
+      update_fn: &copy_alerts_to_s3/2
+    ]
+  end
+
+  @doc """
+   Find alerts that qualify as a bus stop closure or
+   stop move, and copy them to our S3 bucket. This function is invoked in Alerts.Cache.Fetcher.handle_info/2 with two arguments, but we only use the first.
+  """
+  @spec copy_alerts_to_s3([%Alert{}], any) :: :ok
+  def copy_alerts_to_s3(bus_alerts, _) do
+    with stop_change_alerts when stop_change_alerts != [] <-
+           Enum.filter(bus_alerts, &(&1.effect in [:stop_closure, :stop_moved])) do
+      Enum.map(stop_change_alerts, & &1.id)
+      |> Enum.sort()
```

Comment:
Maybe not - would the cache key match if the same alerts were sorted differently (I assumed no)? How confident should I be that the previous functions will maintain alerts in a consistent sort order (I assumed not very, but I might be wrong there)?

Reply:
Yes, you're right that the cache key wouldn't match. However, this also means that you won't re-write the alerts to S3 if the contents of the alert changes.

Reply:
Yep!! I'm using `ttl: :timer.hours(24)` with the understanding that the cached value will expire then, and this will end up re-writing the alerts the next day. 

This reflects that I don't need to capture _very_ timely alert updates, but does expose me to the risk that an alert will make a meaningful change < 24h before concluding or being removed from the V3 API...